openhands/runtime/remote/runtime.py

```diff
@@ -0,0 +1,424 @@
+import asyncio
+import os
+import ssl
+import tempfile
+import uuid
+from typing import Any, Optional, Type
+from zipfile import ZipFile
+
+import aiohttp
+import aiohttp.client_exceptions
+import tenacity
+
+from openhands.core.config import AppConfig
+from openhands.core.logger import openhands_logger as logger
+from openhands.events import EventStream
+from openhands.events.action import (
+    BrowseInteractiveAction,
+    BrowseURLAction,
+    CmdRunAction,
+    FileReadAction,
+    FileWriteAction,
+    IPythonRunCellAction,
+)
+from openhands.events.action.action import Action
+from openhands.events.observation import (
+    ErrorObservation,
+    NullObservation,
+    Observation,
+)
+from openhands.events.serialization import event_to_dict, observation_from_dict
+from openhands.events.serialization.action import ACTION_TYPE_TO_CLASS
+from openhands.runtime.builder.remote import RemoteRuntimeBuilder
+from openhands.runtime.plugins import PluginRequirement
+from openhands.runtime.runtime import Runtime
+from openhands.runtime.utils.runtime_build import build_runtime_image
+
+DEFAULT_RETRY_EXCEPTIONS = [
+    ssl.SSLCertVerificationError,
+    aiohttp.ClientError,
+    aiohttp.client_exceptions.ContentTypeError,
+    aiohttp.client_exceptions.ClientConnectorCertificateError,
+    ssl.SSLCertVerificationError,
+    asyncio.TimeoutError,
+]
+
+
+class RemoteRuntime(Runtime):
+    """This runtime will connect to a remote od-runtime-client."""
+
+    port: int = 60000  # default port for the remote runtime client
+
+    def __init__(
+        self,
+        config: AppConfig,
+        event_stream: EventStream,
+        sid: str = 'default',
+        plugins: list[PluginRequirement] | None = None,
+    ):
+        super().__init__(config, event_stream, sid, plugins)
+        if self.config.sandbox.api_hostname == 'localhost':
+            self.config.sandbox.api_hostname = 'api.all-hands.dev/v0/runtime'
+            logger.warning(
+                'Using localhost as the API hostname is not supported in the RemoteRuntime. Please set a proper hostname.\n'
+                'Setting it to default value: api.all-hands.dev/v0/runtime'
+            )
```

Comment:
ah, "localhost" was the default value in the config - I just want to throw this warning so we can just set a default value for `RemoteRuntime`.

And when people want to use localhost, they should probably go directly with using `EventStreamRuntime`, rather than `RemoteRuntime`?